### PR TITLE
feat(react-email): No forced file extensions on slugs

### PR DESCRIPTION
**Pull request by gabrielmfern**
## What was the issue?

Users were being forced into always having a `.tsx` at the end of the path
for the preview app, this fixes that, meaning that if the user navigates around
between emails there won't be any `.tsx`, but if the user opens the URL
with an existing email and the right file extension, it will render correctly as well.

The latter remark from above is important because it means that we don't really
have to update the docs again to match these paths but the paths are still
going to look better for user experience.

## How can I test to make sure it's fixed?

1. Run `npx tsx ../../packages/react-email/src/cli/index.ts dev` inside of `./apps/demo`
2. Open http://localhost:3000/preview/airbnb-review.tsx and verify that it works
3. Open http://localhost:3000/preview/aribnb-review and verify it also works
4. Navigate between emails and verify that they don't have any `.tsx` file extension on the URL

### CODE REVIEW

**Comment by bukinoshita** [comment]
Should we actually support extensions in the URL? How hard would be to not support it?

**Comment by bukinoshita** [comment]
When having `.tsx` in the url for example, the sidebar item is not active

**Comment by gabrielmfern** [comment]
> Should we actually support extensions in the URL? How hard would be to not support it?

I don't think it would be hard but we'd need to change the links on the docs again through another PR maybe. Wdyt about that?

File: packages/react-email/src/actions/render-email-by-path.tsx [hunk 1]
```diff
@@ -19,11 +17,12 @@ export type EmailRenderingResult =
       error: ErrorObject;
     };
 
-export const renderEmailBySlug = async (
-  emailSlug: string,
+/**
+ * Renders an email file based on the slug.
+ */
```
Comment (vcapretz) [inline packages/react-email/src/actions/render-email-by-path.tsx:22]: unnecessary comment imo

**Review by vcapretz — DISMISSED** [review]

File: packages/react-email/src/actions/render-email-by-path.tsx (hunk 1 above)
Comment (gabrielmfern) [reply packages/react-email/src/actions/render-email-by-path.tsx:22]: yeah LOL, I think I started writing it and forgot about writing the rest of it

**Review by vcapretz — APPROVED** [review]

**Comment by gabrielmfern** [comment]
Thank you so much @vcapretz!

**Pull request merged**